### PR TITLE
[Bullet] Kinematics Features

### PR DESCRIPTION
**Pull request by Lobotuerk**
KinematicsFeatures detected by ignition.
Will rebase after https://github.com/ignitionrobotics/ign-physics/pull/136 gets merged.

### CODE REVIEW

**Review by Blast545 — APPROVED** [review]

File: bullet/src/KinematicsFeatures.cc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+{
+  FrameData3d data;
+
+  // The feature system should never send us the world ID.
+  if (_id.IsWorld())
+  {
+    ignerr << "Given a FrameID belonging to the world. This should not be "
+           << "possible! Please report this bug!\n";
+    assert(false);
+    return data;
```
[the remark this answers is not in the document]
Comment (Lobotuerk) [reply bullet/src/KinematicsFeatures.cc:20]: Not sure, the other implementations use the same, so i left it like that https://github.com/ignitionrobotics/ign-physics/blob/main/dartsim/src/KinematicsFeatures.cc#L34

File: bullet/src/KinematicsFeatures.cc [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  data.pose.translation() = eigenVec;
+
+  // Get frame velocities
+  // Note: only support revolute/fixed joints
+  btVector3 vel = btVector3(0, 0, 0);
+  btVector3 omega = btVector3(0, 0, 0);
+  btMatrix3x3 matChildToThis = btMatrix3x3().getIdentity();
+  btVector3 rThisToChild = btVector3(0, 0, 0);
+  auto linkIndex = linkInfo->linkIndex;
+  while (linkIndex != -1)
```
[the remark this answers is not in the document]
Comment (Lobotuerk) [reply bullet/src/KinematicsFeatures.cc]: Agree, changed on d1d4b94

File: bullet/src/KinematicsFeatures.cc [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  data.pose.linear() = eigenMat;
+  data.pose.translation() = eigenVec;
+
+  // Get frame velocities
+  // Note: only support revolute/fixed joints
+  btVector3 vel = btVector3(0, 0, 0);
+  btVector3 omega = btVector3(0, 0, 0);
+  btMatrix3x3 matChildToThis = btMatrix3x3().getIdentity();
+  btVector3 rThisToChild = btVector3(0, 0, 0);
+  auto linkIndex = linkInfo->linkIndex;
```
[the remark this answers is not in the document]
Comment (Lobotuerk) [reply bullet/src/KinematicsFeatures.cc]: Updated on d1d4b94

File: bullet/src/KinematicsFeatures.cc [hunk 4]
```diff
@@ -45,39 +46,39 @@ FrameData3d KinematicsFeatures::FrameDataRelativeToWorld(
...
-    vel = matThisToParent * vel;
-
-    // Info for next computation
-    matChildToThis = matThisToParent;
-    rThisToChild = link.m_cachedRVector;
-
-    // Move on to parent
-    linkIndex = link.m_parent;
-  }
+  // while (linkIndex != -1)
```
Comment (Blast545) [inline bullet/src/KinematicsFeatures.cc:50]: Please add here a TO-DO explaining why this is left here commented

**Review by Blast545 — APPROVED** [review]
Minor suggestion, otherwise LGTM

**Pull request merged**